error[E0277]: `PhantomPinned` cannot be unpinned
  --> tests/ui/pin_project/add-pinned-field.rs:23:16
   |
23 |     is_unpin::<Foo>(); //~ ERROR E0277
   |                ^^^ within `__Foo<'_>`, the trait `Unpin` is not implemented for `PhantomPinned`
   |
   = note: consider using the `pin!` macro
           consider using `Box::pin` if you need to access the pinned value outside of the current scope
note: required because it appears within the type `__Foo<'_>`
  --> tests/ui/pin_project/add-pinned-field.rs:10:8
   |
10 | struct Foo {
   |        ^^^
note: required for `Foo` to implement `Unpin`
  --> tests/ui/pin_project/add-pinned-field.rs:10:8
   |
 8 | #[pin_project]
   | -------------- type parameter would need to implement `Unpin`
 9 | #[add_pinned_field]
10 | struct Foo {
   |        ^^^
   = help: consider manually implementing `Unpin` to avoid undesired bounds
note: required by a bound in `is_unpin`
  --> tests/ui/pin_project/add-pinned-field.rs:6:16
   |
 6 | fn is_unpin<T: Unpin>() {}
   |                ^^^^^ required by this bound in `is_unpin`

error[E0277]: `PhantomPinned` cannot be unpinned
  --> tests/ui/pin_project/add-pinned-field.rs:24:16
   |
24 |     is_unpin::<Bar>(); //~ ERROR E0277
   |                ^^^ within `__Bar<'_>`, the trait `Unpin` is not implemented for `PhantomPinned`
   |
   = note: consider using the `pin!` macro
           consider using `Box::pin` if you need to access the pinned value outside of the current scope
note: required because it appears within the type `__Bar<'_>`
  --> tests/ui/pin_project/add-pinned-field.rs:17:8
   |
17 | struct Bar {
   |        ^^^
note: required for `Bar` to implement `Unpin`
  --> tests/ui/pin_project/add-pinned-field.rs:17:8
   |
16 | #[pin_project]
   | -------------- type parameter would need to implement `Unpin`
17 | struct Bar {
   |        ^^^
   = help: consider manually implementing `Unpin` to avoid undesired bounds
note: required by a bound in `is_unpin`
  --> tests/ui/pin_project/add-pinned-field.rs:6:16
   |
 6 | fn is_unpin<T: Unpin>() {}
   |                ^^^^^ required by this bound in `is_unpin`
