error[E0793]: reference to field of packed struct is unaligned
  --> tests/ui/pin_project/unaligned_references.rs:17:13
   |
17 |     let _ = &a.f; //~ ERROR reference to packed field is unaligned
   |             ^^^^
   |
   = note: this struct is 1-byte aligned, but the type of this field may require higher alignment
   = note: creating a misaligned reference is undefined behavior (even if that reference is never dereferenced)
   = help: copy the field contents to a local variable, or replace the reference with a raw pointer and use `read_unaligned`/`write_unaligned` (loads and stores via `*p` must be properly aligned even when using raw pointers)

error[E0793]: reference to field of packed struct is unaligned
  --> tests/ui/pin_project/unaligned_references.rs:20:13
   |
20 |     let _ = &b.f; //~ ERROR reference to packed field is unaligned
   |             ^^^^
   |
   = note: this struct is at most 2-byte aligned, but the type of this field may require higher alignment
   = note: creating a misaligned reference is undefined behavior (even if that reference is never dereferenced)
   = help: copy the field contents to a local variable, or replace the reference with a raw pointer and use `read_unaligned`/`write_unaligned` (loads and stores via `*p` must be properly aligned even when using raw pointers)
