error: unexpected token: ()
  --> $DIR/invalid.rs:10:18
   |
10 |         #[project()] //~ ERROR unexpected token
   |                  ^^

error: unexpected token: (foo)
  --> $DIR/invalid.rs:17:18
   |
17 |         #[project(foo)] //~ ERROR unexpected token
   |                  ^^^^^

error: unexpected token: ()
  --> $DIR/invalid.rs:24:18
   |
24 |         #[project()] //~ ERROR unexpected token
   |                  ^^

error: unexpected token: (foo)
  --> $DIR/invalid.rs:33:18
   |
33 |         #[project(foo)] //~ ERROR unexpected token
   |                  ^^^^^

error: unexpected token: foo
  --> $DIR/invalid.rs:42:15
   |
42 |     #[project(foo)] //~ ERROR unexpected token
   |               ^^^

error: duplicate #[project] attribute
  --> $DIR/invalid.rs:56:9
   |
56 |         #[project] //~ ERROR duplicate #[project] attribute
   |         ^^^^^^^^^^

error: duplicate #[project_ref] attribute
  --> $DIR/invalid.rs:64:9
   |
64 |         #[project_ref] //~ ERROR duplicate #[project_ref] attribute
   |         ^^^^^^^^^^^^^^

error: duplicate #[project_replace] attribute
  --> $DIR/invalid.rs:72:9
   |
72 |         #[project_replace] //~ ERROR duplicate #[project_replace] attribute
   |         ^^^^^^^^^^^^^^^^^^

error: attributes `project_ref` and `project` are mutually exclusive
  --> $DIR/invalid.rs:79:9
   |
79 |         #[project]
   |         ^^^^^^^^^^

error: attributes `project` and `project_replace` are mutually exclusive
  --> $DIR/invalid.rs:88:9
   |
88 |         #[project_replace] //~ ERROR are mutually exclusive
   |         ^^^^^^^^^^^^^^^^^^

error: attributes `project_ref` and `project_replace` are mutually exclusive
  --> $DIR/invalid.rs:96:9
   |
96 |         #[project_replace] //~ ERROR are mutually exclusive
   |         ^^^^^^^^^^^^^^^^^^

error: attributes `project_ref` and `project` are mutually exclusive
   --> $DIR/invalid.rs:103:9
    |
103 |         #[project]
    |         ^^^^^^^^^^

error: attributes `project` and `project_replace` are mutually exclusive
   --> $DIR/invalid.rs:112:9
    |
112 |         #[project_replace] //~ ERROR are mutually exclusive
    |         ^^^^^^^^^^^^^^^^^^

error: attributes `project_ref` and `project_replace` are mutually exclusive
   --> $DIR/invalid.rs:120:9
    |
120 |         #[project_replace] //~ ERROR are mutually exclusive
    |         ^^^^^^^^^^^^^^^^^^

error: attributes `project_ref` and `project` are mutually exclusive
   --> $DIR/invalid.rs:127:9
    |
127 |         #[project]
    |         ^^^^^^^^^^

error: attributes `project` and `project_replace` are mutually exclusive
   --> $DIR/invalid.rs:136:9
    |
136 |         #[project_replace] //~ ERROR are mutually exclusive
    |         ^^^^^^^^^^^^^^^^^^

error: attributes `project_ref` and `project_replace` are mutually exclusive
   --> $DIR/invalid.rs:144:9
    |
144 |         #[project_replace] //~ ERROR are mutually exclusive
    |         ^^^^^^^^^^^^^^^^^^

error: duplicate #[project] attribute
   --> $DIR/invalid.rs:149:5
    |
149 |     #[project] //~ ERROR duplicate #[project] attribute
    |     ^^^^^^^^^^

error: duplicate #[project_ref] attribute
   --> $DIR/invalid.rs:153:5
    |
153 |     #[project_ref] //~ ERROR duplicate #[project_ref] attribute
    |     ^^^^^^^^^^^^^^

error: duplicate #[project_replace] attribute
   --> $DIR/invalid.rs:157:5
    |
157 |     #[project_replace] //~ ERROR duplicate #[project_replace] attribute
    |     ^^^^^^^^^^^^^^^^^^

error: duplicate #[project] attribute
   --> $DIR/invalid.rs:161:5
    |
161 |     #[project] //~ ERROR duplicate #[project] attribute
    |     ^^^^^^^^^^

error: duplicate #[project_ref] attribute
   --> $DIR/invalid.rs:165:5
    |
165 |     #[project_ref] //~ ERROR duplicate #[project_ref] attribute
    |     ^^^^^^^^^^^^^^

error: duplicate #[project_replace] attribute
   --> $DIR/invalid.rs:169:5
    |
169 |     #[project_replace] //~ ERROR duplicate #[project_replace] attribute
    |     ^^^^^^^^^^^^^^^^^^

error: duplicate #[project] attribute
   --> $DIR/invalid.rs:177:9
    |
177 |         #[project] //~ ERROR duplicate #[project] attribute
    |         ^^^^^^^^^^

error: duplicate #[project_ref] attribute
   --> $DIR/invalid.rs:181:9
    |
181 |         #[project_ref] //~ ERROR duplicate #[project_ref] attribute
    |         ^^^^^^^^^^^^^^

error: duplicate #[project_replace] attribute
   --> $DIR/invalid.rs:185:9
    |
185 |         #[project_replace] //~ ERROR duplicate #[project_replace] attribute
    |         ^^^^^^^^^^^^^^^^^^
